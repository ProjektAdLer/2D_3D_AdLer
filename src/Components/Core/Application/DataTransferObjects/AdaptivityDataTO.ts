export class AdaptivityDataTO {
  introText: string;
  shuffleTask: boolean;
  tasks: AdaptivityTaskTO[];
}

export class AdaptivityTaskTO {
  taskId: number;
  taskTitle: string;
  taskOptional: boolean;
  requieredDifficulty: number;
  questions: AdaptivityQuestionTO[];
}

export class AdaptivityQuestionTO {
  questionType: string;
  questionId: number;
  questionDifficulty: number;
  questionText: string;
  questionAnswers: AdaptivityAnswersTO[];
  trigger: AdaptivityTriggerTO[];
}

export class AdaptivityAnswersTO {
  answerId: number;
  answerText?: string;
  answerImage?: string;
}

export class AdaptivityTriggerTO {
  triggerId: number;
  triggerType: string;
  triggerCondition: string;
  triggerAction: AdaptivityActionTO;
}

export class AdaptivityActionTO {
  actionType: string;
  actionData: string | number;
}
