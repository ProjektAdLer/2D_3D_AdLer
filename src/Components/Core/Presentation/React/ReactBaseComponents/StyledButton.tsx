import React from "react";

interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  shape?: "square" | "freefloatleft" | "freefloatcenter";
  color?: "default" | "success";
}
export default function StyledButton({
  shape = "square",
  color = "default",
  children,
  className,
  ...rest
}: React.DetailedHTMLProps<ButtonProps, HTMLButtonElement>) {
  const buttonConfig = {
    // Background Colors
    default: "bg-adlerblue",
    success: "bg-adlergreen",

    // Shapes
    square: "justify-center p-1 lg:w-16 md:w-14 sm:w-12 aspect-square",
    freefloatleft: "p-1",
    freefloatcenter: "p-1 justify-center",
  };
  return (
    <button
      className={
        className +
        " " +
        `flex items-center px-2 py-1 text-sm font-black text-white text-shadow border-b-4 border-r-4 rounded-lg hover:cursor-pointer lg:text-xl active:border-transparent border-adlerdarkblue ${buttonConfig[color]} ${buttonConfig[shape]}`
      }
      {...rest}
    >
      {children}
    </button>
  );
}
