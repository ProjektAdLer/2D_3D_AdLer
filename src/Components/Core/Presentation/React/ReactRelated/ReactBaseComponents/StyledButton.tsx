import React from "react";
import tailwindMerge from "../../../Utils/TailwindMerge";
import { AdLerUIComponent } from "src/Components/Core/Types/ReactTypes";

export type StyledButtonColor =
  | "default"
  | "success"
  | "pressed"
  | "locked"
  | "highlight";
export type StyledButtonShape =
  | "square"
  | "freeFloatLeft"
  | "freeFloatCenter"
  | "freeFloatCenterNoPadding"
  | "smallSquare";

interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  shape?: StyledButtonShape;
  color?: StyledButtonColor;
  disabled?: boolean;
  icon?: string;
  containerClassName?: string;
}

export default function StyledButton({
  shape = "square",
  color = "default",
  disabled = false,
  icon,
  children,
  className,
  containerClassName = "",
  ...rest
}: AdLerUIComponent<React.DetailedHTMLProps<ButtonProps, HTMLButtonElement>>) {
  const buttonConfig = {
    // Background Colors
    default: "bg-buttonbgblue",
    success: "bg-adlergreen",
    pressed: "bg-buttonpressedblue",
    locked: "bg-adlerbuttonlocked",
    highlight:
      "bg-adlerdarkblue text-buttonbgblue hover:text-adlerdarkblue border-buttonbgblue",

    // Shapes
    square:
      "justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square",
    freeFloatLeft: "px-2 py-1",
    freeFloatCenter: "flex px-2 py-1 justify-center",
    freeFloatCenterNoPadding: "flex justify-center",
    smallSquare: "justify-center px-2 py-1 aspect-square",
  };

  return (
    <div className={containerClassName}>
      <button
        disabled={disabled}
        className={tailwindMerge(
          className,
          buttonConfig[shape],
          disabled
            ? "box-border text-adlerdeactivatedtext bg-adlerbuttonlocked flex items-center text-sm rounded-lg lg:text-xl font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-white overflow-hidden"
            : tailwindMerge(
                "flex items-center text-sm rounded-lg hover:cursor-pointer  landscape:hover:bg-adleryellow  lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent active:bg-adleryellow text-adlerdarkblue font-regular  border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer",
                buttonConfig[color],
              ),
        )}
        {...rest}
      >
        {icon && (
          <img
            alt=""
            className={icon && children ? "h-6 lg:h-8 pr-2" : "h-8 lg:h-12"}
            src={icon}
          ></img>
        )}
        {children}
      </button>
    </div>
  );
}
