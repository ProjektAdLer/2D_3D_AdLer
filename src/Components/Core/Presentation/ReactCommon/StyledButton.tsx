import React from "react";

export default function StyledButton({
  children,
  className,
  ...rest
}: React.HTMLAttributes<HTMLButtonElement>) {
  return (
    <button
      className={
        className +
        "container-button hover:cursor-pointer text-sm lg:text-xl border-b-4 border-r-4 border-adlerdarkblue active:border-adlergold m-auto py-1 px-2 bg-adlerblue rounded-lg font-black text-white"
      }
      {...rest}
    >
      {children}
    </button>
  );
}
