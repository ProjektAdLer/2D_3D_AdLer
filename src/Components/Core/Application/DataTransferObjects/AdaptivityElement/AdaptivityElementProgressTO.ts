import { AdaptivityDataTO } from "../AdaptivityDataTO";
import AdaptivityElementTaskProgressTO from "./AdaptivityElementTaskProgressTO";

export default class AdaptivityElementProgressTO extends AdaptivityDataTO {
  isCompleted: boolean;
  override tasks: AdaptivityElementTaskProgressTO[];
}
