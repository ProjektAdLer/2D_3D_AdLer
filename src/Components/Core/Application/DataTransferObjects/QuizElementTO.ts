import { StyledButtonColor } from "./../../Presentation/React/ReactRelated/ReactBaseComponents/StyledButton";
import { ComponentID } from "../../Domain/Types/EntityTypes";

export class SubmittedAnswersTO {
  questionID: number;
  selectedAnswerIndexes: number[];
  allAnswerIndexes: number[];
}

export class CorrectAnswersTO {
  questionID: number;
  answerIndexes: number[];
}

export class EvaluationAnswerTO {
  questionID: number;
  evaluation: Map<number, StyledButtonColor>;
}

export class AnswerTO {
  answerIndex: number;
  questionTitle?: string;
  answerText: string;
  answerImage?: string;
}

export class Question {
  questionID: string;
  questionTitle?: string;
  questionText: string;
  questionImage?: string;
  questionPoints: number;
  questionAnswers: AnswerTO[];
}

export class AdaptivityReference {
  comment: string;
  reference: ComponentID;
}

export class AdaptivityResponse {
  result: string | Question | AdaptivityReference;
}

export class AdaptivityData {
  adaptivityType: string;
  resultType: string;
  condition: string;
  result: AdaptivityResponse;
}

export class TOAdaptivityQuestionTO {
  questionID: number;
  questionTitle?: string;
  questionText: string;
  questionImage?: string;
  questionPoints: number;
  questionAnswers: AnswerTO[];
}

export class AdaptivityContentsTO {
  shuffleQuestions: boolean;
  questions: TOAdaptivityQuestionTO[];
}
