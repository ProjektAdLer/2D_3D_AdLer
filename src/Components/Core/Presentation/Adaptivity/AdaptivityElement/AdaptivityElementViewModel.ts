import { AdaptivityElementQuestionDifficultyTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import Observable from "src/Lib/Observable";
import { StyledButtonColor } from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

export interface AdaptivityAnswer {
  answerIndex: number;
  answerText: string;
  isSelected: boolean;
}

export interface AdaptivityQuestion {
  questionID: number;
  questionText: string;
  questionAnswers: AdaptivityAnswer[];
  isRequired: boolean;
  isCompleted: boolean;
  difficulty: AdaptivityElementQuestionDifficultyTypes;
  isMultipleChoice: boolean;
}

export interface AdaptivityTask {
  taskID: number;
  taskTitle: string;
  questions: AdaptivityQuestion[];
  isCompleted: boolean;
  requiredDifficulty: number;
}

export interface AdaptivityElementContent {
  elementName: string;
  introText: string;
  tasks: AdaptivityTask[];
}

export default class AdaptivityElementViewModel {
  isOpen: Observable<boolean> = new Observable<boolean>(false);
  showFeedback: Observable<boolean> = new Observable<boolean>(false);
  contentData: Observable<AdaptivityElementContent> =
    new Observable<AdaptivityElementContent>();
  currentTaskID: Observable<number | null> = new Observable<number | null>(
    null
  );
  currentQuestionID: Observable<number | null> = new Observable<number | null>(
    null
  );
  footerText: Observable<string> = new Observable<string>("");
}
