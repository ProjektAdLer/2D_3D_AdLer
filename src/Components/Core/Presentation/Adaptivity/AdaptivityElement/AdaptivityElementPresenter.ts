import { AdaptivityElementQuestionTypes } from "./../../../Domain/Types/Adaptivity/AdaptivityElementQuestionTypes";
import IAdaptivityElementPresenter from "./IAdaptivityElementPresenter";
import AdaptivityElementViewModel, {
  AdaptivityElementContent,
  AdaptivityQuestion,
  AdaptivityTask,
  AdaptivityAnswer,
  AdaptivityHint,
  AdaptivityHintAction,
} from "./AdaptivityElementViewModel";
import AdaptivityElementProgressTO from "../../../Application/DataTransferObjects/AdaptivityElement/AdaptivityElementProgressTO";
import bind from "bind-decorator";
import { AdaptivityElementQuestionDifficultyTypes } from "../../../Domain/Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import AdaptivityElementProgressUpdateTO from "../../../Application/DataTransferObjects/AdaptivityElement/AdaptivityElementProgressUpdateTO";
import { AdaptivityElementStatusTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementStatusTypes";
import AdaptivityElementTaskProgressTO from "src/Components/Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementTaskProgressTO";
import AdaptivityElementQuestionProgressTO from "src/Components/Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementQuestionProgressTO";
import AdaptivityElementAnswersTO from "src/Components/Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementAnswerTO";
import AdaptivityElementTriggerTO from "src/Components/Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementTriggerTO";
import { AdaptivityElementTriggerConditionTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementTriggerConditionTypes";
import AdaptivityElementHintTO from "src/Components/Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementHintTO";
import i18next from "i18next";

export default class AdaptivityElementPresenter
  implements IAdaptivityElementPresenter
{
  constructor(private viewModel: AdaptivityElementViewModel) {
    viewModel.currentTask.subscribe(() => this.setFooterBreadcrumbs());
    viewModel.currentQuestion.subscribe(() => this.setFooterBreadcrumbs());
    viewModel.contentData.subscribe(() => this.setFooterBreadcrumbs());
  }

  onAdaptivityElementLoaded(
    adaptivityElementProgressTO: AdaptivityElementProgressTO
  ): void {
    this.setContentData(adaptivityElementProgressTO);
    this.setFooterBreadcrumbs();

    this.viewModel.elementID.Value = adaptivityElementProgressTO.id;
    this.viewModel.model.Value = adaptivityElementProgressTO.model;
    this.viewModel.currentTask.Value = null;
    this.viewModel.currentQuestion.Value = null;

    this.viewModel.isOpen.Value = true;
  }

  onAdaptivityElementAnswerEvaluated(
    adaptivityElementProgressUpdateTO: AdaptivityElementProgressUpdateTO
  ): void {
    const updatedTask = this.viewModel.contentData.Value.tasks.find(
      (task) =>
        task.taskID === adaptivityElementProgressUpdateTO.taskInfo.taskId
    )!;

    const taskStatus = adaptivityElementProgressUpdateTO.taskInfo.taskStatus;
    if (taskStatus === AdaptivityElementStatusTypes.Correct) {
      updatedTask.isCompleted = true;
      updatedTask.hasBeenCompleted = true;
    } else if (taskStatus === AdaptivityElementStatusTypes.Incorrect)
      updatedTask.isCompleted = false;
    else updatedTask.isCompleted = null;

    const updatedQuestion = updatedTask.questions.find(
      (question) =>
        question.questionID ===
        adaptivityElementProgressUpdateTO.questionInfo.questionId
    )!;

    const questionStatus =
      adaptivityElementProgressUpdateTO.questionInfo.questionStatus;
    if (questionStatus === AdaptivityElementStatusTypes.Correct)
      updatedQuestion.isCompleted = true;
    else if (questionStatus === AdaptivityElementStatusTypes.Incorrect)
      updatedQuestion.isCompleted = false;
    else updatedQuestion.isCompleted = null;

    updatedQuestion.questionAnswers.forEach((answer) => {
      answer.isSelected = false;
    });

    this.viewModel.showFeedback.Value = true;
  }

  onAdaptivityElementUserHintInformed?(
    adaptivityElementHintTO: AdaptivityElementHintTO
  ): void {
    const hint = {
      hintID: adaptivityElementHintTO.hintID,
      showOnIsWrong: adaptivityElementHintTO.showOnIsWrong,
      hintAction: {
        hintActionType: adaptivityElementHintTO.hintAction.hintActionType,
        idData: adaptivityElementHintTO.hintAction.idData,
        textData: adaptivityElementHintTO.hintAction.textData,
      } as AdaptivityHintAction,
    } as AdaptivityHint;
    this.viewModel.selectedHint.Value = hint;
  }

  @bind
  private setFooterBreadcrumbs(): void {
    let newFooterText = this.viewModel.contentData.Value.elementName;

    if (this.viewModel.currentTask.Value !== null) {
      newFooterText += " \u2794 " + this.viewModel.currentTask.Value.taskTitle;

      if (this.viewModel.currentQuestion.Value !== null) {
        switch (this.viewModel.currentQuestion.Value.difficulty) {
          case AdaptivityElementQuestionDifficultyTypes.easy:
            newFooterText +=
              " \u2794 " + i18next.t("easyQuestion", { ns: "learningElement" });
            break;
          case AdaptivityElementQuestionDifficultyTypes.medium:
            newFooterText +=
              " \u2794 " +
              i18next.t("normalQuestion", { ns: "learningElement" });
            break;
          case AdaptivityElementQuestionDifficultyTypes.hard:
            newFooterText +=
              " \u2794 " + i18next.t("hardQuestion", { ns: "learningElement" });
            break;
        }
      }
    }

    this.viewModel.footerText.Value = newFooterText;
  }

  private setContentData(
    adaptivityElementProgressTO: AdaptivityElementProgressTO
  ): void {
    const newTasks: AdaptivityTask[] = this.mapTasks(
      adaptivityElementProgressTO.tasks
    );

    this.viewModel.contentData.Value = {
      elementName: adaptivityElementProgressTO.elementName,
      introText: adaptivityElementProgressTO.introText,
      tasks: newTasks,
    } as AdaptivityElementContent;
  }

  private mapTasks(tasks: AdaptivityElementTaskProgressTO[]): AdaptivityTask[] {
    return tasks.map((task) => {
      return {
        taskID: task.taskId,
        taskTitle: task.taskTitle,
        questions: this.mapQuestions(
          task.questions,
          task.requiredDifficulty,
          task.taskOptional
        ),
        isCompleted: task.isCompleted,
        hasBeenCompleted: task.isCompleted !== null ? task.isCompleted : false,
        requiredDifficulty: task.requiredDifficulty,
        isRequired: !task.taskOptional,
      } as AdaptivityTask;
    });
  }

  private sortQuestionsByDifficulty(
    questions: AdaptivityElementQuestionProgressTO[]
  ) {
    questions.sort(
      (
        a: AdaptivityElementQuestionProgressTO,
        b: AdaptivityElementQuestionProgressTO
      ) => {
        return a.questionDifficulty - b.questionDifficulty;
      }
    );
  }

  private mapQuestions(
    questions: AdaptivityElementQuestionProgressTO[],
    requiredDifficulty: AdaptivityElementQuestionDifficultyTypes,
    isTaskOptional: boolean
  ): AdaptivityQuestion[] {
    this.sortQuestionsByDifficulty(questions);
    return questions.map((question) => {
      // TODO: this is not completely correct yet
      // What happens if the required difficulty is something like 86?
      const isRequired =
        requiredDifficulty === question.questionDifficulty && !isTaskOptional;

      const isMultipleChoice =
        question.questionType ===
        AdaptivityElementQuestionTypes.multipleResponse;

      return {
        questionID: question.questionId,
        questionText: question.questionText,
        questionAnswers: this.mapAnswers(question.questionAnswers),
        isRequired: isRequired,
        isCompleted: question.isCompleted,
        difficulty: question.questionDifficulty,
        isMultipleChoice: isMultipleChoice,
        hints: this.mapAdaptivityTriggers(question.triggers),
      } as AdaptivityQuestion;
    });
  }

  private mapAnswers(
    answers: AdaptivityElementAnswersTO[]
  ): AdaptivityAnswer[] {
    return answers.map((answer) => {
      return {
        answerIndex: answer.answerId,
        answerText: answer.answerText,
        isSelected: false,
      } as AdaptivityAnswer;
    });
  }

  private mapAdaptivityTriggers(
    triggers: AdaptivityElementTriggerTO[]
  ): AdaptivityHint[] {
    if (!triggers.length) return [];

    return triggers.map((trigger) => {
      return {
        hintID: trigger.triggerId,
        showOnIsWrong:
          trigger.triggerCondition ===
          AdaptivityElementTriggerConditionTypes.incorrect,
        hintAction: {
          hintActionType: trigger.triggerAction.actionType,
          idData: trigger.triggerAction.idData,
          textData: trigger.triggerAction.textData,
        } as AdaptivityHintAction,
      } as AdaptivityHint;
    });
  }
}
