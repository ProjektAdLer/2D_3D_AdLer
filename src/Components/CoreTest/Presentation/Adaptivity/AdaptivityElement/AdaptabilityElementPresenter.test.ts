import AdaptivityElementProgressTO from "../../../../Core/Application/DataTransferObjects/AdaptivityElement/AdaptivityElementProgressTO";
import { EvaluationAnswerTO } from "../../../../Core/Application/DataTransferObjects/QuizElementTO";
import AdaptivityElementPresenter from "../../../../Core/Presentation/Adaptivity/AdaptivityElement/AdaptivityElementPresenter";
import AdaptivityElementViewModel, {
  AdaptivityElementContent,
} from "../../../../Core/Presentation/Adaptivity/AdaptivityElement/AdaptivityElementViewModel";
import { StyledButtonColor } from "../../../../Core/Presentation/React/ReactRelated/ReactBaseComponents/StyledButton";

describe("AdaptivityElementPresenter", () => {
  let systemUnderTest: AdaptivityElementPresenter;
  let viewModel: AdaptivityElementViewModel;

  beforeEach(() => {
    viewModel = new AdaptivityElementViewModel();
    systemUnderTest = new AdaptivityElementPresenter(viewModel);
  });

  test("onAdaptivityElementLoaded sets isOpen to true", () => {
    const adaptivityElementProgressTO: AdaptivityElementProgressTO = {
      isCompleted: false,
      tasks: [
        {
          taskId: 1,
          taskTitle: "TestTaskTitle",
          taskOptional: false,
          questions: [
            {
              questionId: 1,
              questionText: "TestQuestionText",
              questionType: "",
              trigger: [],
              questionAnswers: [
                {
                  answerId: 1,
                  answerText: "TestAnswerText",
                },
              ],
              questionDifficulty: 1,
              isCompleted: false,
            },
          ],
          requiredDifficulty: 1,
          isCompleted: false,
        },
      ],
      introText: "",
      shuffleTask: false,
      elementName: "",
    };

    systemUnderTest.onAdaptivityElementLoaded(adaptivityElementProgressTO);

    expect(systemUnderTest["viewModel"].isOpen.Value).toBe(true);
  });

  test("createFooterBreadcrumbs returns the correct string of names", () => {
    const contentData: AdaptivityElementContent = {
      elementName: "TestElement",
      introText: "",
      tasks: [
        {
          taskID: 1,
          taskTitle: "TestTask",
          questions: [
            {
              questionID: 1,
              questionText: "",
              questionAnswers: [],
              isRequired: false,
              isCompleted: false,
              difficulty: 100,
            },
          ],
          isCompleted: false,
          requiredDifficulty: 100,
        },
      ],
    };
    systemUnderTest["viewModel"].contentData.Value = contentData;
    systemUnderTest["viewModel"].currentTaskID.Value = 1;
    systemUnderTest["viewModel"].currentQuestionID.Value = 1;

    systemUnderTest["setFooterBreadcrumbs"]();

    // TODO: extend expected result when question title is implemented
    expect(viewModel.footerText.Value).toBe("TestElement => TestTask");
  });

  test("createContentData returns the correct content data", () => {
    const adaptivityElementProgressTO: AdaptivityElementProgressTO = {
      isCompleted: false,
      tasks: [
        {
          taskId: 1,
          taskTitle: "TestTaskTitle",
          taskOptional: false,
          questions: [
            {
              questionId: 1,
              questionText: "TestQuestionText",
              questionType: "",
              trigger: [],
              questionAnswers: [
                {
                  answerId: 1,
                  answerText: "TestAnswerText",
                },
              ],
              questionDifficulty: 1,
              isCompleted: false,
            },
          ],
          requiredDifficulty: 1,
          isCompleted: false,
        },
      ],
      introText: "",
      shuffleTask: false,
      elementName: "TestName",
    };

    systemUnderTest["setContentData"](adaptivityElementProgressTO);

    expect(viewModel.contentData.Value).toStrictEqual({
      elementName: "TestName",
      introText: "",
      tasks: [
        {
          taskID: 1,
          taskTitle: "TestTaskTitle",
          isCompleted: false,
          requiredDifficulty: 1,
          questions: [
            {
              questionID: 1,
              questionText: "TestQuestionText",
              isRequired: true,
              isCompleted: false,
              difficulty: 1,
              questionAnswers: [
                {
                  answerIndex: 1,
                  answerText: "TestAnswerText",
                  isSelected: false,
                },
              ],
            },
          ],
        },
      ],
    });
  });

  test("onAdaptivityElementSubmitted should set evaluation in viewModel", () => {
    const mappedValues: Map<number, StyledButtonColor> = new Map([
      [1, "success"],
      [2, "success"],
    ]);
    const evaluate: EvaluationAnswerTO = {
      questionID: 1,
      evaluation: mappedValues,
    };

    systemUnderTest.onAdaptivityElementSubmitted(evaluate);

    expect(systemUnderTest["viewModel"].evaluation.Value).toEqual(
      evaluate.evaluation
    );
  });
});
