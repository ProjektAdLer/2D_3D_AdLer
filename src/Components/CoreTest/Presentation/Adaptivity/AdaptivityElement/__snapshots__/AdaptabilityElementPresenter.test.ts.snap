// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptivityElementPresenter createFooterBreadcrumbs returns the correct string of names 1`] = `"TestElement => TestTask => Mittlere Frage"`;
